tensorflow/core/profiler/internal/gpu/rocm_tracer.h

```diff
@@ -0,0 +1,256 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
```

Comment:
playing devils advocate here

is there way we can have a common `gpu_tracer.h`? 
* Then for the implementation we can have `cupti_wrapper.cc` on the cuda side, and `rocm_tracer.cc` on the rocm side
* This is would also potentially allow us to keep a common `device_tracer.cc` file

Think it would be very difficult to pull this off, but odds are this question will come up during the upstream review process.

Reply:
@deven-amd we certainly can, and probably should extract common logic for upstreaming purpose...

I just revised HCC patch ( https://github.com/RadeonOpenCompute/hcc/pull/1314 ) after some discussion with @scchan and @jeffdaily . Now working on getting `rocm_tracer.cc` to incorporate mGPU book-keeping logic. Will then address comments from you here.